cyipopt/scipy_interface.py

```diff
@@ -84,19 +101,34 @@ def __init__(self, fun, args=(), kwargs=None, jac=None, hess=None,
         self.kwargs = kwargs or {}
         self._constraint_funs = []
         self._constraint_jacs = []
+        self._constraint_funs_with_jacs = []
+        self._constraint_hessians = []
+        self._constraint_dims = np.asarray(con_dims)
         self._constraint_args = []
         self._constraint_kwargs = []
+        self.last_con_values = [(None, None) for _ in range(len(constraints))]
```

Comment:
Is this a public attribute typically supplied as part of the `scipy.optimize.minimize` API? Should this be public or private given that other similar attributes for this class are private?

Reply:
No, it's only needed to cache the function value and the evaluated Jacobian for each constraint and used inside the `__evaluate_con_fun_with_jac()` method. So I agree that it should rather be private.